app/models/foreman_docker/service/actions/container/pull.rb

```diff
@@ -0,0 +1,28 @@
+module ForemanDocker
+  module Service
+    module Actions
+      module Container
+        class Pull < ::Actions::EntryAction
+          def plan(container)
+            action_subject(container)
+            plan_self(:container => container.id)
+          end
+
+          def run
+            container = ::Container.find(input[:container])
+            container.compute_resource.create_image(:fromImage => container.repository_pull_url)
```

Comment:
It's highly variable, hours even, depending on the bandwidth and the size of the image. Usually it's a 1GB download, after it's downloaded the image is added to docker pretty instantaneously.